FHEM/00_SIGNALduino.pm

```diff
@@ -2921,7 +2923,26 @@ sub SIGNALduino_Attr(@) {
 		return "Note, please use other userReadings names.\nReserved names from $name are: cc1101_config, cc1101_config_ext, cc1101_patable"
 			if ($aVal =~ /cc1101_(?:config(?:_ext)?|patable)(?:\s|{)/);
 	}
- 	return ;
+	elsif ($aName eq "rfmode")	# change receive mode
+	{
+		my $oldAttrib = AttrVal($name, 'rfmode', 'SlowRF');
+
+		if ($aVal ne $oldAttrib) {
+			$hash->{logMethod}->($name, 3, "$name: Attr, $aName set to $aVal (please check protocollist)");
+			if ($aVal ne 'SlowRF') {
+				my ($manu, $adr) = cc1101_manspecific::Register($aVal);         # return array with cc1101 manufacturer specific´s register settings
+				for my $i (0...scalar(@{$manu})-1) {
+					$hash->{logMethod}->($name, 5, "$name: Attr, $aName - write adress " . @{$adr}[$i] . " -> " . @{$manu}[$i]);
```

Comment:
Du gehst davon aus, dass @manu und @adr die gleiche Menge an Werten haben.

Immerhin, ein kürzerer array @manu als @adr führt hier zu keinem Problem :) Aber sobald @manu mehr Werte hat, bekommen wir einen Zugriff  auf einen Wert in @adr den es nicht gibt.

Reply:
Das verstehe ich und ist uns bewußt.

Wir sind davon ausgegangen das nach der Zusammenstellung aller Register nur die geschrieben werden, welche sich verändern. Das zusammengestellt, kommt auf die Adressen.

Mir geht durch den kopf, wenn wir nun pro Protokolleintrag einmal das Register hinterlegen mit Werten und dazu die Adresse, so würde man deinen genannten Fall erschlagen wenn dieser mal hinzukommt, ODER man sieht, das man das Register nicht benötigt, so könnte man es einzeln heraustrennen mit der Adresse.

Vorschlag:
```
RegisterAdr      => '00,01',
RegisterVal       => '01,2D',
```

?

Reply:
Ich würde 
Registers =>  '00' => '01, '01' => '2D' hinterlegen. So ist gut erkennbar, wekches Register welchen Wert erhalten soll.

Reply:
Nachdem ich mir das heute nochmal angesehen habe, so habe ich die Register den Sensoren zugeordnet.

Die Register werden in einem Array gespeichert welches nun unabhängig ist. Enthalten ist die Adresse mit dessen Wert.
Das sollte nun deinen Vorstellungen entsprechen sowohl auch die Variante, nur an einer Stelle die spezifischen Informationen zu pflegen.